gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -181,12 +183,38 @@ private ManagedChannel createSingleChannel() throws IOException {
     int port = Integer.parseInt(endpoint.substring(colon + 1));
     String serviceAddress = endpoint.substring(0, colon);
 
-    ManagedChannelBuilder builder =
-        ManagedChannelBuilder.forAddress(serviceAddress, port)
-            .intercept(headerInterceptor)
-            .intercept(metadataHandlerInterceptor)
-            .userAgent(headerInterceptor.getUserAgentHeader())
-            .executor(executor);
+    // TODO(hzyi): Use NettyChannelBuilder when maxInboundMetadataSize is specified to unblock Spanner.
+    // Change to ManagedChannelBuilder when https://github.com/grpc/grpc-java/issues/4050 is
+    // resolved.
+    ManagedChannelBuilder builder;
+    if (maxInboundMetadataSize != null) {
+      try {
+        Class.forName("io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder");
```

Comment:
What's the point in this if there is a compile dependency on `grpcNettyShaded`?

Reply:
It's to allow for users to swap out grpc-netty-shaded and use grpc-netty instead of they want.